packages/core/test/scp-cf/destination-accessor.spec.ts

```diff
@@ -1069,4 +1022,192 @@ describe('destination-accessor', () => {
+        mockThingsForCombinations();
+
+        const requestSpy = jest.spyOn(
+          destinationService,
+          'fetchSubaccountDestinations'
+        );
+        const actualProvider = await fetchDestination(
+          subscriberUserJwt,
+          subscriberFirst
+        );
+
+        expect(requestSpy).toHaveBeenNthCalledWith(
+          1,
+          'https://destination.example.com',
+          subscriberServiceToken,
+          expect.anything()
+        );
+        expect(requestSpy).toHaveBeenNthCalledWith(
+          2,
+          'https://destination.example.com',
+          providerServiceToken,
+          expect.anything()
+        );
+        expect(actualProvider!.url).toBe(subscriberDestination.URL);
+      });
+    });
+
+    describe('providerUserToken x  {alwaysSubscriber,alwaysProvider,subscriberFirst}', () => {
+      it('providerUserToken x alwaysSubscriber: should return null since the token does not match subscriber', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(
+          providerUserJwt,
+          alwaysSubscriber
+        );
+        assertNothingCalledAndNullFound(mocks, actual);
+      });
+
+      it('providerUserToken x alwaysProvider: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(providerUserJwt, alwaysProvider);
+        assertSubscriberNotCalledAndProviderFound(mocks, actual!);
+      });
+
+      it('providerUserToken x subscriberFirst: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(providerUserJwt, subscriberFirst);
+        assertSubscriberNotCalledAndProviderFound(mocks, actual!);
+      });
+    });
+
+    describe('no UserToken x {alwaysSubscriber,alwaysProvider,subscriberFirst}', () => {
+      it('noUserToken && alwaysSubscriber: should return null since the token does not match subscriber', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(undefined, alwaysSubscriber);
+        assertNothingCalledAndNullFound(mocks, actual);
+      });
+
+      it('noUserToken && alwaysProvider: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(undefined, alwaysProvider);
+        assertSubscriberNotCalledAndProviderFound(mocks, actual!);
+      });
+
+      it('noUserToken x subscriberFirst: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
```

Comment:
"send" (also above)

Reply:
what is the difference between `x` and `&&` above? Maybe the descriptions can be slightly improved?